### ISSUE TITLE
test(core): `TestExchange_DoNotStoreHistoric` flake

### ISSUE THREAD

**Issue by renaynay** [celestiaorg/celestia-node#3931]
```
--- FAIL: TestExchange_DoNotStoreHistoric (12.25s)
    network.go:48: tearing down testnode
    exchange_test.go:139: 
        	Error Trace:	/home/runner/work/celestia-node/celestia-node/core/exchange_test.go:139
        	            				/opt/hostedtoolcache/go/1.23.2/x64/src/runtime/asm_amd64.s:1700
        	Error:      	Received unexpected error:
        	            	deliverTxSub was cancelled (reason: CometBFT exited)
        	Test:       	TestExchange_DoNotStoreHistoric
2024-11-11T14:11:06.120Z	ERROR	core	core/listener.go:195	listener: received block with unexpected chain ID: expected wrong-chain-rpc, received private
FAIL
```

**Comment by Wondertan** [comment]
Got this on https://github.com/celestiaorg/celestia-node/actions/runs/11842973768/job/33002886332
```
 --- FAIL: TestExchange_DoNotStoreHistoric (12.25s)
    network.go:48: tearing down testnode
    exchange_test.go:138: 
        	Error Trace:	/home/runner/work/celestia-node/celestia-node/core/exchange_test.go:138
        	            				/opt/hostedtoolcache/go/1.23.2/x64/src/runtime/asm_amd64.s:1700
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unavailable desc = error reading from server: EOF
        	Test:       	TestExchange_DoNotStoreHistoric
2024-11-14T18:13:19.617Z	ERROR	core	core/listener.go:169	listener: resubscribe	{"err": "failed to subscribe: context canceled"}
2024-11-14T18:13:30.687Z	ERROR	core	core/listener.go:194	listener: received block with unexpected chain ID: expected wrong-chain-rpc, received private
FAIL
```

**Comment by cristaloleg** [comment]
@Wondertan is this one-time failure or something constant?

**Comment by Wondertan** [comment]
I saw it only once @cristaloleg, after the fix was merged

**Comment by walldiss** [comment]
No longer relevant
